src/libraries/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -106,156 +107,114 @@ internal static class UriHelper
             return true;
         }
 
-        internal static string EscapeString(
-            string stringToEscape, // same name as public API
-            bool checkExistingEscaped, ReadOnlySpan<bool> unreserved, char forceEscape1 = '\0', char forceEscape2 = '\0')
+        internal static string EscapeString(string stringToEscape, bool checkExistingEscaped, IndexOfAnyValues<char> noEscape)
         {
             ArgumentNullException.ThrowIfNull(stringToEscape);
 
-            if (stringToEscape.Length == 0)
-            {
-                return string.Empty;
-            }
-
-            // Get the table of characters that do not need to be escaped.
-            Debug.Assert(unreserved.Length == 0x80);
-            scoped ReadOnlySpan<bool> noEscape;
-            if ((forceEscape1 | forceEscape2) == 0)
-            {
-                noEscape = unreserved;
-            }
-            else
-            {
-                Span<bool> tmp = stackalloc bool[0x80];
-                unreserved.CopyTo(tmp);
-                tmp[forceEscape1] = false;
-                tmp[forceEscape2] = false;
-                noEscape = tmp;
-            }
+            Debug.Assert(!noEscape.Contains('%'), "Need to treat % specially; it should be part of any escaped set");
 
-            // If the whole string is made up of ASCII unreserved chars, just return it.
-            Debug.Assert(!noEscape['%'], "Need to treat % specially; it should be part of any escaped set");
-            int i = 0;
-            char c;
-            for (; i < stringToEscape.Length && (c = stringToEscape[i]) <= 0x7F && noEscape[c]; i++) ;
-            if (i == stringToEscape.Length)
+            int indexOfFirstToEscape = stringToEscape.AsSpan().IndexOfAnyExcept(noEscape);
+            if (indexOfFirstToEscape < 0)
             {
+                // If the whole string is made up of ASCII unreserved chars, just return it.
```

Comment:
Nit: This (existing) comment feels a little out of place, since `noEscape` is an arbitrary input and at least by name isn't limited to ASCII unreserved chars.  Should the parameter be named differently?